openedx/features/calendar_sync/__init__.py

```diff
@@ -0,0 +1,16 @@
+"""
+Calendar syncing Course dates with a User.
+"""
+
+
+def get_calendar_event_id(user, block_key):
+    """
+    Creates a unique event id based on a user and a course block key
+
+    Parameters:
+        user (User): The user requesting a calendar event
+        block_key (str): The block key containing the date for the calendar event
+    Returns:
+        event id (str)
+    """
+    return user.username + '.' + block_key
```

Comment:
This needn't block this PR, but blocks can have multiple "kinds" of dates, yeah? I think we mostly focus on the `due` type, but it's technically possible to have more?

Reply:
Great catch. updated